homeassistant/components/soma/cover.py

```diff
@@ -21,39 +38,175 @@ async def async_setup_entry(
+                f'Error while opening the cover ({self.name}): {response["msg"]}'
+            )
+
+    def stop_cover_tilt(self, **kwargs):
+        """Stop the cover tilt."""
+        response = self.api.stop_shade(self.device["mac"])
+        if is_api_response_success(response):
+            # Set cover position to some value where up/down are both enabled
+            self.set_position(50)
+        else:
+            raise HomeAssistantError(
+                f'Error while stopping the cover ({self.name}): {response["msg"]}'
+            )
+
+    def set_cover_tilt_position(self, **kwargs):
+        """Move the cover tilt to a specific position."""
+        # 0 -> Closed down (api: 100)
+        # 50 -> Fully open (api: 0)
+        # 100 -> Closed up (api: -100)
+        target_api_position = 100 - ((kwargs[ATTR_TILT_POSITION] / 50) * 100)
+        response = self.api.set_shade_position(self.device["mac"], target_api_position)
+        if is_api_response_success(response):
+            self.set_position(kwargs[ATTR_TILT_POSITION])
+        else:
+            raise HomeAssistantError(
+                f'Error while setting the cover position ({self.name}): {response["msg"]}'
+            )
+
+    async def async_update(self):
+        """Update the entity with the latest data."""
+        try:
+            _LOGGER.debug("Soma Tilt Update")
+            response = await self.hass.async_add_executor_job(
+                self.api.get_shade_state, self.device["mac"]
+            )
+            if not self.api_is_available:
+                self.api_is_available = True
+                _LOGGER.info("Connection to SOMA Connect succeeded")
+        except RequestException:
+            if self.api_is_available:
+                _LOGGER.warning("Connection to SOMA Connect failed")
+                self.api_is_available = False
+            return
+
+        if not is_api_response_success(response):
+            if self.is_available:
+                self.is_available = False
+                _LOGGER.warning(
+                    f'Device is unreachable ({self.name}). Error while fetching the state: {response["msg"]}'
+                )
+            return
+
+        if not self.is_available:
+            self.is_available = True
+            _LOGGER.info(f"Device {self.name} is now reachable")
+
+        api_position = int(response["position"])
+
+        if "closed_upwards" in response.keys():
+            self.current_position = 50 + ((api_position * 50) / 100)
+        else:
+            self.current_position = 50 - ((api_position * 50) / 100)
+
+
+class SomaShade(SomaEntity, CoverEntity):
+    """Representation of a Soma Shade device."""
+
+    @property
+    def device_class(self):
+        """Return the class of this device."""
+        return DEVICE_CLASS_SHADE
+
+    @property
+    def supported_features(self):
+        """Flag supported features."""
+        return SUPPORT_OPEN | SUPPORT_CLOSE | SUPPORT_STOP | SUPPORT_SET_POSITION
+
```

Comment:
```suggestion
 
    _attr_device_class = DEVICE_CLASS_SHADE
    _attr_supported_features = SUPPORT_OPEN | SUPPORT_CLOSE | SUPPORT_STOP | SUPPORT_SET_POSITION

```

Reply:
Didn't know about that, thanks for that suggestion!
Fixed